src/components/Nav/NavTop.jsx

```diff
@@ -0,0 +1,202 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import Modal from 'react-modal';
+import styled from 'styled-components';
+import { FaRegHeart } from 'react-icons/fa';
+import { FaUserCircle } from 'react-icons/fa';
+import { FaChevronDown } from 'react-icons/fa';
+import { FaSearch } from 'react-icons/fa';
+import SubscribeModal from '../SubscribeModal/SubscribeModal';
+import ProfileModal from './ProfileModal';
+
+function NavTop() {
+  const [subscribeModalOpen, setSubscribeModalOpen] = useState(false);
+  const [profileModalOpen, setProfileModalOpen] = useState(false);
+  const navigate = useNavigate();
+
+  return (
+    <Header>
+      <Nav>
+        <WrapInfo>
+          <Logo>IOI</Logo>
+          <Subscribe onClick={() => setSubscribeModalOpen(!subscribeModalOpen)}>
```

Comment:
```suggestion
          <Subscribe onClick={() => setSubscribeModalOpen(prev => !prev)}>
```
아래 이와 같이 적용된 부분 모두 함수형 업데이트 형태로 수정해주세요!